gh-129515: Clarify syntax error messages for conditional expressions (#129880)
python/cpython: Misc/NEWS.d/next/Core_and_Builtins/2025-02-09-11-30-38.gh-issue-129515.3L3vmo.rst

> Clarify syntax error messages for conditional expressions if statement

Reviewer: This is kind of a new feature so a What's New entry would also be great. It could be put in the same section where we improved `ValueError` messages, and it would also be good to actually show some examples (`body` and `orelse` parts are not necessarily known).

Reviewer: In addition, you could also link to the grammar specification and use the `:keyword:` role for `if`.

Author: I tried to update the "What's New" section, but I'm not the best at writing in English.

> Clarify syntax error messages for conditional expressions if statement
> specified before ``if`` or after ``else`` keywords.

Reviewer: ```suggestion
Clarify syntax error messages for conditional expressions when a statement
is specified before an :keyword:`if` or after an :keyword:`else` keyword.
```

Author: Thanks!
